source/common/grpc/google_grpc_creds_impl.cc

```diff
@@ -0,0 +1,144 @@
+#include "common/grpc/google_grpc_creds_impl.h"
+
+#include "envoy/api/v2/core/grpc_service.pb.h"
+#include "envoy/grpc/google_grpc_creds.h"
+#include "envoy/registry/registry.h"
+
+#include "common/config/datasource.h"
+
+namespace Envoy {
+namespace Grpc {
+
+/**
+ * Default implementation of Google Grpc Credentials Factory
+ * Uses ssl creds if available, or defaults to insecure channel.
```

Comment:
Should there be a TODO here to take all of the statically defined credentials in the `GoogleGrpc` config and turn them into channel/call credentials?